streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -65,7 +65,7 @@
 import static org.apache.kafka.streams.state.internals.TimeOrderedKeyValueBufferChangelogDeserializationHelper.deserializeV3;
 import static org.apache.kafka.streams.state.internals.TimeOrderedKeyValueBufferChangelogDeserializationHelper.duckTypeV2;
 
-public final class InMemoryTimeOrderedKeyValueBuffer<K, V> implements TimeOrderedKeyValueBuffer<K, V> {
+public final class InMemoryTimeOrderedKeyValueBuffer<K, V, T> implements TimeOrderedKeyValueBuffer<K, V, Change<V>> {
```

Comment:
Could be good to rename this to `InMemoryTimeOrderedKeyValueChangeBuffer` to contrast with `RocksDBTimeOrderedKeyValueBuffer` (not a change buffer) in your next PR, but I don't feel too strongly.